HW03_Graphs.Rmd

```diff
@@ -0,0 +1,174 @@
+
+To practice importing files with different R packages, I decided to divide this homework in three sections: 
+
+- utils 
+- readr 
+- readxl
+
+## utils package
+
+For this section, I'm going to recreate circular dichroism (*CD*) and catalytic activities graphs, for that purpose, I created a couple of *txt* files (available in data), one with data separated by slashes "/", and the other with tabulate-separated values.
+
+#### Circular Dichroism
+
+Let's start with CD spectra.
+
+This one is the original graph:
+![Circular Dicroism](images/dicro.JPG)
+
+And here's the code for its recreation using ggplot2:
+
+```{r CD, echo = TRUE}
+
+#I choose some colors and shapes for each species beforehand from a colorblind-friendly palette. To simplify the scripts, here are some vectors for it
+colorg <- c("#E69F00","#D55E00")
+shapeg <- c(25, 23)
+
+dc_data <- read.table("data/dichroism.txt", 
+                      header = TRUE, 
+                      sep = "/", 
+                      stringsAsFactors = FALSE)
+
+#CD data, Mean Residue Ellipticity ([theta]), is usually divided by 1000 to facilitate visualization
+dc_data$Ellip_mod <- dc_data$Ellipticity / 1000
+
+ggplot(data = dc_data, mapping = aes(x = Wavelength, y = Ellip_mod, color = Species, shape = Species)) +
+  geom_line(size = 2) +
+  geom_hline(yintercept = 0, size = 1) + #to add the reference line in y = 0
+  scale_color_manual(values = colorg) +
+  labs(title = "CD spectra",
+       x = "Wavelenght (nm)",
+       y = expression(paste("[",theta,"] x 1000 (deg cm "^{2}, " dmol "^{-1}, ")"))) + #{} are used to write things in superscripts, `expression()` also serves to add greek letters
+  coord_cartesian(xlim = c(190, 260), ylim = c(-12, 25), expand = FALSE) + #To zoom the data accordingly
+  #To specify the spacing between ticks
+  scale_x_continuous(breaks = seq(190, 260, 10)) +
+  scale_y_continuous(breaks = seq(-15, 25, 5))
+```
+
+What this data tells us is that both enzymes have the secondary structure of an alpha/beta protein.
+
+#### Catalytic Activity
+
+As with any other enzymes, it is important the determine how good they are at performing catalysis.
+
+The original graph was this:
+
+![Catalytic activity](images/activity.JPG)
+
+And this is the R recreation:
+
+```{r catalytic activity, echo = TRUE}
+cat_data <- read.delim("data/catalysis.txt", 
+                       header = TRUE, 
+                       stringsAsFactors = FALSE)
+
+ggplot(data = cat_data, mapping = aes(x = Substrate, y = Specific.activity, fill = Species, shape = Species)) +
+    geom_errorbar(aes(ymin = Specific.activity-SD, ymax = Specific.activity+SD), width = 0.08, alpha = 0.4) + #A good chance to practice the Stat chapter of DataCamp
+  geom_point(color = "white", size = 3) +
```

Comment:
Clever manipulation of the outline color for the points to get it to match your original graph!